toonz/sources/toonz/startuppopup.cpp

```diff
@@ -0,0 +1,889 @@
+  TApp::instance()->getCurrentScene()->getScene()->setScenePath(
+      TFilePath(m_pathFld->getPath()) +
+      TFilePath(m_nameFld->text().trimmed().toStdWString()));
+  TDimensionD size =
+      TDimensionD(m_widthFld->getValue(), m_heightFld->getValue());
+  TDimension res = TDimension(m_xRes, m_yRes);
+  double fps     = m_fpsFld->getValue();
+  TApp::instance()
+      ->getCurrentScene()
+      ->getScene()
+      ->getProperties()
+      ->getOutputProperties()
+      ->setFrameRate(fps);
+  TApp::instance()->getCurrentScene()->getScene()->getCurrentCamera()->setSize(
+      size);
+  TApp::instance()->getCurrentScene()->getScene()->getCurrentCamera()->setRes(
+      res);
+  // this one is debatable - should the scene be saved right away?
+  // IoCmd::saveScene();
+  // this makes sure the scene viewers update to the right fps
+  TApp::instance()->getCurrentScene()->notifySceneSwitched();
+
+  hide();
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::updateProjectCB() {
+  m_updating = true;
+  m_projectPaths.clear();
+  m_projectsCB->clear();
+
+  TFilePath sandboxFp = TProjectManager::instance()->getSandboxProjectFolder() +
+                        "sandbox_otprj.xml";
+  m_projectPaths.push_back(sandboxFp);
+  m_projectsCB->addItem("sandbox");
+
+  std::vector<TFilePath> prjRoots;
+  TProjectManager::instance()->getProjectRoots(prjRoots);
+  for (int i = 0; i < prjRoots.size(); i++) {
+    TFilePathSet fps;
+    TSystem::readDirectory_Dir_ReadExe(fps, prjRoots[i]);
+
+    TFilePathSet::iterator it;
+    for (it = fps.begin(); it != fps.end(); ++it) {
+      TFilePath fp(*it);
+      if (TProjectManager::instance()->isProject(fp)) {
+        m_projectPaths.push_back(
+            TProjectManager::instance()->projectFolderToProjectPath(fp));
+        m_projectsCB->addItem(QString::fromStdString(fp.getName()));
+      }
+    }
+  }
+  int i;
+  for (i = 0; i < m_projectPaths.size(); i++) {
+    if (TProjectManager::instance()->getCurrentProjectPath() ==
+        m_projectPaths[i]) {
+      m_projectsCB->setCurrentIndex(i);
+      break;
+    }
+  }
+  m_pathFld->setPath(TApp::instance()
+                         ->getCurrentScene()
+                         ->getScene()
+                         ->getProject()
+                         ->getScenesPath()
+                         .getQString());
+  m_updating = false;
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::onProjectChanged(int index) {
+  if (m_updating) return;
+  TFilePath projectFp = m_projectPaths[index];
+
+  TProjectManager::instance()->setCurrentProjectPath(projectFp);
+
+  IoCmd::newScene();
+  m_pathFld->setPath(TApp::instance()
+                         ->getCurrentScene()
+                         ->getScene()
+                         ->getProject()
+                         ->getScenesPath()
+                         .getQString());
```

Comment:
I think I fixed that.  I also added enter in the name field animating the click for create scene